src/generic_ui/scripts/user.ts

```diff
@@ -41,46 +40,61 @@ module UI {
       }
       this.name = profile.name;
       this.imageData = profile.imageData || UI.DEFAULT_USER_IMG;
-      this.isOnline = profile.isOnline;
     }
 
     // Returns two strings, where each matches an array name in model.contacts.
     // Does not use an enum because we need a string value, and typescript
     // enums evaluate to numbers.
```

Comment:
You can get the string of a typescript enum easily, e.g. 
```
enum Color {Red = 1, Green, Blue};
var colorName: string = Color[Color.Red];
```
enums have saved us from string-errors that lead you to spending time debugging. They also save time when refactoring and allow you to change names with some security (typechecker tells you when you mess up). I think we should prefer `Color[Color.Red]` over "red" if we need to index by a string.